Source/NSGridView.m

```diff
@@ -0,0 +1,1280 @@
+              NSDebugLog(@"_owningRow = %@", _owningRow);
+            }
+          if ([coder containsValueForKey: @"NSGrid_owningColumn"])
+            {
+              _owningColumn = [coder decodeObjectForKey: @"NSGrid_owningColumn"]; // weak
+              NSDebugLog(@"_owningColumn = %@", _owningColumn);
+            }
+          if ([coder containsValueForKey: @"NSGrid_xPlacement"])
+            {
+              _xPlacement = [coder decodeIntegerForKey: @"NSGrid_xPlacement"];
+            }      
+          if ([coder containsValueForKey: @"NSGrid_yPlacement"])
+            {
+              _yPlacement = [coder decodeIntegerForKey: @"NSGrid_yPlacement"];
+            }  
+          if ([coder containsValueForKey: @"NSGrid_alignment"])
+            {
+              _rowAlignment = [coder decodeIntegerForKey: @"NSGrid_alignment"];
+            }      
+        }
+      else
+        {
+          [self setContentView: [coder decodeObject]];
+          ASSIGN(_mergeHead, [coder decodeObject]);
+          _owningRow = [coder decodeObject]; // weak
+          _owningColumn = [coder decodeObject]; // weak
+          [coder decodeValueOfObjCType: @encode(NSInteger)
+                                    at: &_xPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_yPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_rowAlignment];
+        }
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_contentView);
+  RELEASE(_mergeHead);
+  [super dealloc];
+}
+
+@end
+
+/// Column ///
+@implementation NSGridColumn
+
++ (void) initialize
+{
+  if (self == [NSGridColumn class])
+    {
+      [self setVersion: 1];
+    }
+}
+
+- (void) setGridView: (NSGridView *)gv
+{
+  _gridView = gv;
+}
+
+- (NSGridView *) gridView
+{
+  return _gridView;
+}
+
+- (NSInteger) numberOfCells
+{
+  NSUInteger ci = [_gridView indexOfColumn: self];
+  NSArray *arr = [_gridView _cellsForColumnAtIndex: ci];
+  return [arr count];
+}
+
+- (NSGridCell *) cellAtIndex:(NSInteger)index
+{
+  NSUInteger ci = [_gridView indexOfColumn: self];
+  NSArray *arr = [_gridView _cellsForColumnAtIndex: ci];
+  return [arr objectAtIndex: index];
+}
+
+- (NSGridCellPlacement) xPlacement
+{
+  return _xPlacement;
+}
+
+- (void) setXPlacement: (NSGridCellPlacement)x
+{
+  _xPlacement = x;
+}
+
+- (CGFloat) width
+{
+  return _width;
+}
+
+- (void) setWidth: (CGFloat)f
+{
+  _width = f;
+}
+
+- (CGFloat) leadingPadding
+{
+  return _leadingPadding;
+}
+
+- (void) setLeadingPadding: (CGFloat)f
+{
+  _leadingPadding = f;
+}
+
+- (CGFloat) trailingPadding
+{
+  return _trailingPadding;
+}
+
+- (void) setTrailingPadding: (CGFloat)f
+{
+  _trailingPadding = f;
+}
+
+- (BOOL) isHidden
+{
+  return _isHidden;
+}
+
+- (void) mergeCellsInRange: (NSRange)range
+{
+  NSLog(@"Method %@ unimplemented.", NSStringFromSelector(_cmd));
```

Comment:
Shouldn't you be calling mergeCellsInHorizontalRange:verticalRange: on the grid view? I really don't know.

Reply:
Done.  I believe I have this correct.